modules/stream-shredder/src/test/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/stream/sinks/ProcessingSpec.scala

```diff
@@ -0,0 +1,187 @@
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and
+ * limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.shredder.stream.sinks
+
+import java.nio.file.Path
+import java.time.Instant
+
+import cats.implicits._
+
+import cats.effect.{Blocker, ContextShift, IO, Timer}
+
+import fs2.io.{file => fs2File}
+import fs2.{Stream, text}
+
+import io.circe.Json
+import io.circe.optics.JsonPath._
+import io.circe.parser.{ parse => parseCirce }
+
+import com.snowplowanalytics.iglu.client.Client
+import com.snowplowanalytics.iglu.core.SchemaKey
+
+import com.snowplowanalytics.snowplow.rdbloader.common.LoaderMessage
+import com.snowplowanalytics.snowplow.rdbloader.common.config.ShredderConfig
+import com.snowplowanalytics.snowplow.rdbloader.common.transformation.Transformed
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.Processing
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.Processing.Windowed
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.sources.{Parsed, file => FileSource}
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.sinks.generic.Record
+import com.snowplowanalytics.snowplow.rdbloader.generated.BuildInfo
+
+import org.specs2.mutable.Specification
+
+class ProcessingSpec extends Specification {
+  import ProcessingSpec._
+  "transform" should {
+    "shred events correctly" in {
+      val (good, bad) = transformTestEvents(resourcePath = "/processing-spec/1/input/events", format = shredFormat)
+
+      val pathsToCheck = List(
+        Transformed.Path.Shredded.Tabular("com.snowplowanalytics.snowplow", "atomic", 1),
+        Transformed.Path.Shredded.Tabular("com.snowplowanalytics.snowplow", "consent_document", 1),
+        Transformed.Path.Shredded.Tabular("com.optimizely", "state", 1)
+      )
+      val expectedTransformedMap = getExpectedTransformedEvents(good, pathsToCheck, 1, LoaderMessage.Format.TSV)
+
+      val expectedBadEvent = getResourceLines("/processing-spec/1/output/bad").head
+
+      bad must have size(1)
+      bad.head.data.value mustEqual expectedBadEvent
+      good must have size(46)
+      // Checks whether transformed events are identical with the expected ones.
+      // Only the paths in "pathsToCheck" are compared to not add
+      // all shredded events to the test folder.
+      good must contain(beLike[(Transformed.Path, Transformed.Data)] {
+        case (path: Transformed.Path.Shredded, data) if pathsToCheck.contains(path) =>
+          expectedTransformedMap(path) must contain(data.value)
+        case (_: Transformed.Path.Shredded, _) => ok
+      })
```

Comment:
I think you are missing `.forall` on this line, or else the `contain` only looks for 1 matching element in the `good` list.

```
      good must contain(beLike[(Transformed.Path, Transformed.Data)] {
        case (path: Transformed.Path.Shredded, data) if pathsToCheck.contains(path) =>
          expectedTransformedMap(path) must contain(data.value)
        case (_: Transformed.Path.Shredded, _) => ok
      }).forall
```

Reply:
Yep, you are right. I will add it 👍